cmd/maya-apiserver/app/server/backup_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,388 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	snapshot "github.com/openebs/maya/pkg/snapshot/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type backupAPIOps struct {
+	req  *http.Request
+	resp http.ResponseWriter
+}
+
+// backupV1alpha1SpecificRequest deals with backup API requests
+func (s *HTTPServer) backupV1alpha1SpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	backupOp := &backupAPIOps{
+		req:  req,
+		resp: resp,
+	}
+
+	switch req.Method {
```

Comment:
should we have a DELETE, as well, to delete backup related CRs and its snapshots?

Reply:
ok